src/datasources/transaction-api/transaction-api.service.ts

```diff
@@ -147,6 +150,69 @@ export class TransactionApi implements ITransactionApi {
     await this.cacheService.deleteByKey(key);
   }
 
+  // TODO: this replicates logic from the CacheFirstDataSource.get method to avoid
+  // implementation of response remapping but we should refactor it to avoid duplication
+  async isSafe(safeAddress: `0x${string}`): Promise<boolean> {
+    const cacheDir = CacheRouter.getIsSafeCacheDir({
+      chainId: this.chainId,
+      safeAddress,
+    });
+
+    const cached = await this.cacheService.get(cacheDir).catch(() => null);
+
+    if (cached != null) {
+      this.loggingService.debug({
+        type: 'cache_hit',
+        ...cacheDir,
+      });
+
+      return cached === 'true';
+    } else {
+      this.loggingService.debug({
+        type: 'cache_miss',
+        ...cacheDir,
+      });
+    }
+
+    const isSafe = await (async (): Promise<boolean> => {
+      try {
+        const url = `${this.baseUrl}/api/v1/safes/${safeAddress}`;
+        const { data } = await this.networkService.get({
+          url,
+        });
+
+        return !!data;
+      } catch (error) {
+        if (
+          error instanceof NetworkResponseError &&
+          // Transaction Service returns 404 when address is not of a Safe
+          error.response.status === 404
+        ) {
+          return false;
+        }
+        throw this.httpErrorFactory.from(this.mapError(error));
+      }
+    })();
+
+    await this.cacheService.set(
+      cacheDir,
+      JSON.stringify(isSafe),
+      // We can indefinitely cache this as an address cannot "un-Safe" itself
+      // and invalidation (clearIsSafe) is called on SAFE_CREATED event
+      TransactionApi.INDEFINITE_EXPIRATION_TIME,
```

Comment:
I'd store this for `INDEFINITE_EXPIRATION_TIME` if `isSafe === true` only. 

Imagine that for some reason (events source/queue malfunction, downtime, etc.) the service does not receive the `SAFE_CREATED` event. There is no problem if the Safe exists, because as you said it cannot be undeployed from the blockchain, but a false negative will be stored forever, as the `SAFE_CREATED` event won't be emitted again. What do you think?

Reply:
Good point. I changed it in 43fb39a6923480a82d15146cb4317eb806821f1f to set it as described, otherwise using the default expiration time.